src/components/Modal/Modal.tsx

```diff
@@ -9,46 +9,35 @@ interface Props {
   children: ReactNode
   onClose: () => void
   className?: string
+  width?: number
+  height?: number | 'auto'
   'data-testid'?: string
 }
 
-export const Modal: FC<Props> = ({ children, onClose, className, 'data-testid': dataTestId }) => {
+export const Modal: FC<Props> = ({
+  children,
+  onClose,
+  className,
+  width,
+  height = 'auto',
+  'data-testid': dataTestId,
+}) => {
   return createPortal(
     <div className="fixed inset-0 flex items-center justify-center z-50" data-testid={dataTestId}>
       {/* Backdrop */}
       <div className="fixed inset-0 bg-[var(--color-bg-100)]/50 backdrop-blur-xs"></div>
 
       {/* Modal Container */}
-      {/* Desktop Modal - Hidden on mobile */}
       <div
         className={cn(
-          'hidden md:block',
           'relative',
-          'bg-[var(--color-bg-80)]',
-          'rounded-[4px]',
-          'overflow-hidden',
-
-          className,
-        )}
-      >
-        {/* Close Button */}
-        <button onClick={onClose} className="absolute top-4 right-4 z-10" data-testid="CloseButton">
-          <Image src="/images/close-button.svg" width={24} height={24} alt="Close" />
-        </button>
-
-        {children}
-      </div>
-
-      {/* Mobile Modal - Hidden on desktop */}
-      <div
-        className={cn(
-          'block md:hidden',
-          'relative w-[380px]',
+          width ? `w-[${width}px]` : 'w-[380px] md:w-[600px]', // Default responsive width,
+          'max-w-[90vw]', // Default Responsive width
+          height !== 'auto' ? `h-[${height}px]` : 'h-auto',
           'max-w-[90vw]',
```

Comment:
There's a redudant `max-w-[90vw]`, need to check which one is needed and rmeove the extra.

Reply:
sorted. Thanks for pointing that out